headerfs/store.go

```diff
@@ -85,14 +85,14 @@ var headerBufPool = sync.Pool{
 }
 
 // headerStore combines a on-disk set of headers within a flat file in addition
-// to a databse which indexes that flat file. Together, these two abstractions
+// to a database which indexes that flat file. Together, these two abstractions
 // can be used in order to build an indexed header store for any type of
 // "header" as it deals only with raw bytes, and leaves it to a higher layer to
 // interpret those raw bytes accordingly.
 //
 // TODO(roasbeef): quickcheck coverage
 type headerStore struct {
-	mtx sync.RWMutex
+	mtx sync.RWMutex // nolint:structcheck, false positive
```

Comment:
Interesting. I removed the lint exception to see what the warning was and got `mtx is unused`. It seems to not like the fact that `headerStore` is embedded in other structs.

Reply:
Yeah, this is a bit annoying. I tried the newest version of `golanglint-ci` but that also still flags it. Added a comment to make clear why it's a false positive.